invertedai_cpp/invertedai/blame_request.cc

```diff
@@ -0,0 +1,169 @@
+#include "blame_request.h"
+
+#include "externals/json.hpp"
+
+using json = nlohmann::json;
+
+namespace invertedai {
+
+BlameRequest::BlameRequest(const std::string &body_str) {
+  this->body_json_ = json::parse(body_str);
+
+  this->location_ = this->body_json_["location"];
+  this->colliding_agents_ =
+      std::make_pair(this->body_json_["colliding_agents"][0],
+                     this->body_json_["colliding_agents"][1]);
+  this->agent_state_history_.clear();
+  for (const auto &elements : this->body_json_["agent_state_history"]) {
+    std::vector<AgentState> agent_states;
+    agent_states.clear();
+    for (const auto &element : elements) {
+      AgentState agent_state = {element[0], element[1], element[2], element[3]};
+      agent_states.push_back(agent_state);
+    }
+    this->agent_state_history_.push_back(agent_states);
+  }
+  this->agent_attributes_.clear();
+  for (const auto &element : this->body_json_["agent_attributes"]) {
+    AgentAttributes agent_attribute = {element[0], element[1], element[2]};
+    this->agent_attributes_.push_back(agent_attribute);
+  }
+  if (this->body_json_["traffic_light_state_history"].is_null()) {
+    this->traffic_light_state_history_ = std::nullopt;
+  } else {
+    std::vector<std::vector<TrafficLightState>> traffic_light_state_history;
+    for (const auto &elements :
+         this->body_json_["traffic_light_state_history"]) {
+      std::vector<TrafficLightState> traffic_light_states;
+      traffic_light_states.clear();
+      for (const auto &element : elements) {
+        TrafficLightState traffic_light_state = {element[0], element[1]};
+        traffic_light_states.push_back(traffic_light_state);
+      }
+      traffic_light_state_history.push_back(traffic_light_states);
+    }
+    this->traffic_light_state_history_ = traffic_light_state_history;
+  }
```

Comment:
I know we use the "auto" keyword in the other request.cc files and this isn't exclusive to this PR in particular but is there a reason we don't use explicit type definitions instead? Is it not guaranteed, for example, that **this->body_json_["agent_state_history"]** returns **std::vector<AgentState>**?